File: COT_TOT_C/input_path/ansible_input_path_C/bug6.c
#include <stdio.h>
#include <string.h>
int buggy_function(int rs, char *data, int freq) {
    char version[]="*";
    char requirement[]="*";
    int parent=1;
    if(parent && strcmp(version,"*")==0 &&
       strcmp(requirement,"*")!=0)
    {
        return rs;
    }
    else if(strcmp(requirement,"*")==0 ||
            strcmp(version,"*")==0)
    {
        printf("Continue\n");
    }
    return rs;
}

File: COT_TOT_PYTHON/agent4_results_tot/ansible_agent4_output_code/bug6.py
"""
Buggy NON-MODULAR snippet from ansible, bug ID 6.
"""
def buggy_function(rs, data, freq=None):
                # In the case we are checking a new requirement on a base requirement (parent != None) we can't accept
                    # version as '*' (unknown version) unless the requirement is also '*'.
                    if parent and version == '*' and requirement != '*':
                        break
                    elif requirement == '*' or version == '*':
                        continue
                version = manifest['version']
            existing[0].add_requirement(to_text(collection_info), requirement)
    return rs